core/src/main/scala/org/allenai/common/cache/QueryCache.scala

```diff
@@ -0,0 +1,66 @@
+package org.allenai.common.cache
+
+import org.allenai.common.Logging
+
+import redis.clients.jedis.{ Jedis, JedisPool }
+
+import spray.json._
+
+/** class holding a Redis cache of query results. This is meant to store any value T where
+  * T <: Object, keyed on string query. If we ever need to run multiple instances within the same application
+  * user neeeds to handle using a shared Jedis Pool.
+  * @param pool the JedisPool that the client should use to serve requests
+  * @param clientPrefix an identifier for the client using this caching mechanism, which will become
+  * part of the cache key (prepended to the actual query)
+  */
+class QueryCache[V: JsonFormat](pool: JedisPool, clientPrefix: String)
```

Comment:
I'd almost call this `JsonQueryCache`, in case we ever want to make a parent class that just stores strings . . .